### DISCUSSION TITLE
Dental panoramic detection: Overfitting problem

### DISCUSSION THREAD

**Discussion by gksruf** [AlexeyAB/darknet#7840] [Q&A]
Hello. I am not an expert in deep learning and seeking for an help. I am trying to customize YOLOv4 with two pathologic lesions on dental panoramic radiographs. Class 1 has total of 455 images (384 training set & 71 test set) and Class 2 has total of 1441 images (1223 training set & 218 test set). All dental panoramic images seems similar in general because people have similar bone structure. However, two different lesions are usually located in the apex of the tooth and they are both pretty small in the large panoramic radiograph. Since labeled lesions are small, we cropped the image when training and tested them. However, the resulting graph looks like below:

![image](https://user-images.githubusercontent.com/12556799/123533886-f6ee4580-d753-11eb-9a14-e6b70c29c034.png)

It overfits very soon. What should I do to fix this problem? Should I try to change masks, anchors, classes, or num in "yolov4.cfg"? Would it help improve the problem? Especially, I am thinking about changing anchors and I wonder how.
Or should I use YOLOv3 instead of v4? or could I change v4 more shallower using darknet? Please help me solve the problem. Thank you and have a great day!

**Comment by stephanecharette** [comment]
Cropped how?  To just the tooth in question?  Note:  https://www.ccoderun.ca/programming/darknet_faq/#crop_training_images

How big are the images?  How big is the decay (or object) you are trying to detect?  What dimensions are you using for the network?

If you have just 2 classes, then you probably should be using yolov4-tiny.  Depending on the image size, object size, and network size, then you may need to enable tiling.  See this page to understand how these 3 dimensions work:  https://www.ccoderun.ca/darkhelp/api/Tiling.html

The problem is not overfitting.  Nor will changing anchors help.  And going from YOLOv3 to YOLOv4 wont help either, with the exception that when you switch between models you change the default network dimensions.

Contact me directly if you'd like additional help:  stephanecharette@gmail.com.
